lib/AST/ASTPrinter.cpp

```diff
@@ -2969,9 +2971,11 @@ void PrintAST::visitSubscriptDecl(SubscriptDecl *decl) {
   }, [&] { // Parameters
     printGenericDeclGenericParams(decl);
     auto params = ArrayRef<AnyFunctionType::Param>();
-    if (decl->hasInterfaceType() && !decl->isInvalid()) {
-      // Walk to the params of the subscript's indices.
-      params = decl->getInterfaceType()->castTo<AnyFunctionType>()->getParams();
+    if (auto type = decl->getInterfaceType()) {
+      if (!decl->isInvalid()) {
+        // Walk to the params of the subscript's indices.
+        params = decl->getInterfaceType()->castTo<AnyFunctionType>()->getParams();
```

Comment:
This should use the `type` from above I think.

Reply:
Good catch, fixed!